src/chromium/crBrowser.ts

```diff
@@ -105,87 +107,103 @@ export class CRBrowser extends platform.EventEmitter implements Browser {
+
+    if (targetInfo.type === 'background_page') {
+      const backgroundPage = new CRPage(session, targetInfo.targetId, context, null, false);
+      this._backgroundPages.set(targetInfo.targetId, backgroundPage);
+      backgroundPage.pageOrError().then(() => {
+        context.emit(Events.CRBrowserContext.BackgroundPage, backgroundPage._page);
+      });
       return;
     }
-    const { context, target } = this._createTarget(targetInfo, session);
-
-    if (CRTarget.isPageType(targetInfo.type)) {
-      target.pageOrError().then(async () => {
-        const page = target._crPage!.page();
-        if (targetInfo.type === 'page') {
-          this._firstPageCallback();
-          context.emit(CommonEvents.BrowserContext.Page, page);
-          const opener = target.opener();
-          if (!opener)
-            return;
-          // Opener page must have been initialized already and resumed in order to
-          // create this popup but there is a chance that not all responses have been
-          // received yet so we cannot use opener._crPage?.page()
-          const openerPage = await opener.pageOrError();
-          if (openerPage instanceof Page && !openerPage.isClosed())
-            openerPage.emit(CommonEvents.Page.Popup, page);
-        } else if (targetInfo.type === 'background_page') {
-          context.emit(Events.CRBrowserContext.BackgroundPage, page);
-        }
+
+    if (targetInfo.type === 'page') {
+      const openerIndex = targetInfo.openerId ? this._popupOpeners.indexOf(targetInfo.openerId) : -1;
+      // When this page is a result of window.open($url) call, we should have it's opener
+      // in the list of popup openers. In this case we know there is an initial
+      // about:blank navigation, followed by a navigation to $url.
+      const hasInitialAboutBlank = openerIndex !== -1;
+      if (openerIndex !== -1)
+        this._popupOpeners.splice(openerIndex, 1);
+      helper.addEventListener(session, 'Page.windowOpen', (event: Protocol.Page.windowOpenPayload) => {
+        // window.open($url) with noopener forces a new browser-initiated window, not
+        // a renderer-initiated one. When url is about:blank, we only get an
+        // initial navigation, and no real navigation to $url.
+        if (event.windowFeatures.includes('noopener') && event.url.startsWith('about:blank'))
+          return;
+        this._popupOpeners.push(targetInfo.targetId);
+      });
+
+      const opener = targetInfo.openerId ? this._crPages.get(targetInfo.openerId) || null : null;
+      const crPage = new CRPage(session, targetInfo.targetId, context, opener, hasInitialAboutBlank);
+      this._crPages.set(targetInfo.targetId, crPage);
+      crPage.pageOrError().then(() => {
+        this._firstPageCallback();
+        context.emit(CommonEvents.BrowserContext.Page, crPage._page);
+        if (opener && opener._initializedPage && !opener._initializedPage.isClosed())
```

Comment:
I mentioned this before, opener._initializedPage may not be there yet even if the opener page has been initialized and resumed. I'd rather keep `const openerPage = await opener.pageOrError();` to be on the safe side.